rwgame/debugstate.cpp

```diff
@@ -172,19 +201,89 @@ DebugState::DebugState(RWGame* game, const glm::vec3& vp, const glm::quat& vd)
 				obj->setSolid(false);
 			}
 		}
-	}, entryHeight));
+	}, kDebugEntryHeight));
 
+	return m;
+}
 
-	// Optional block if the player is in a vehicle
-	auto player = game->getPlayer()->getCharacter();
-	auto cv = player->getCurrentVehicle();
-	if(cv) {
-		m->addEntry(Menu::lambda("Flip vehicle", [=] {
-			cv->setRotation(cv->getRotation() * glm::quat(glm::vec3(0.f, glm::pi<float>(), 0.f)));
-		}, entryHeight));
+Menu* DebugState::createVehicleMenu()
+{
+	CharacterObject* player = nullptr;
+	if (game->getPlayer()) {
+		player = game->getPlayer()->getCharacter();
+	}
+
+	Menu* m = new Menu(2);
+	m->offset = kDebugMenuOffset;
+
+	m->addEntry(Menu::lambda("Back", [=] {
+		this->enterMenu(createDebugMenu());
+	}, kDebugEntryHeight));
+
+#define SPAWN_VEHICLE(name, id) \
+	m->addEntry(Menu::lambda("Add " name, [=] { \
+		auto playerRot = player->getRotation(); \
+		auto spawnPos = player->getPosition(); \
+		spawnPos += playerRot * glm::vec3(0.f, 3.f, 0.f); \
+		auto spawnRot = glm::quat(glm::vec3(0.f, 0.f, glm::roll(playerRot) + glm::half_pi<float>())); \
+		auto car = game->getWorld()->createVehicle(id, spawnPos, spawnRot); \
+		RW_UNUSED(car);\
+	}, kDebugEntryHeight))
+
+	SPAWN_VEHICLE("Yakuza", 136);
+	SPAWN_VEHICLE("Landstalker", 90);
+	SPAWN_VEHICLE("Bobcat", 112);
+	SPAWN_VEHICLE("Banshee", 119);
+	SPAWN_VEHICLE("Rhino", 122);
+	SPAWN_VEHICLE("Barracks", 123);
+	SPAWN_VEHICLE("Rumpo", 130);
+	SPAWN_VEHICLE("Columbian", 138);
+	SPAWN_VEHICLE("Dodo", 126);
+	SPAWN_VEHICLE("Speeder", 142);
```

Comment:
I also have a new car spawner in my taxi branch:
https://github.com/JayFoxRox/openrw/pull/11/commits/8f4cc97f149f4ca6d57ee16057f487ee78a7c244

I need those vehicles + the spawner code is cleaner as it uses the DebugState spawn function